### DISCUSSION TITLE
Compound Indexes in Monger Query mq/hint

### DISCUSSION THREAD

**Discussion by sstraust** [michaelklishin/monger#237] [Q&A] [answered]
Hi! I can't figure out how to get [monger.query/hint](http://reference.clojuremongodb.info/monger.query.html) working with compound indexes in monger query.

i.e., I have a compound index
```
  {
    v: 2,
    key: { sourceName: 1, 'published-date': -1 },
    name: 'sourceName_1_published-date_-1'
  }
```
in my mongo database, 

and I'd like to provide a [hint](https://www.mongodb.com/docs/v5.0/reference/operator/meta/hint/) to ensure this index is used in my db query (issued from  clojure).

I think kind of a fundamentalish issue is that index maps are _ordered_ (i.e. {sourceName: 1, published-date: -1} represents a different index than {published-date: -1, sourceName: -1}), but clojure maps are _not ordered_ so I'm not sure what I'm supposed to pass in or how to organize it to get around this.

I tried messing around with it for a bit, but I'm not really sure what to do or how to get it to work.

Thanks! -- Sammy

p.s. I read through the docs as best I could, and couldn't find anything, but if there is documentation on this, point me to it, and that would solve my issue! :D

**Comment by michaelklishin** [comment, marked as answer]
@sstraust consider using discussions instead of issues for questions. Clojure does have [ordered maps](https://clojuredocs.org/clojure.core/sorted-map) and anything that implements the key map/collection/seq interfaces can be used with the `hint` function. Including ordered maps.

**Reply by sstraust** [reply]
thanks! at the time I really thought it was an issue based on my lack of knowledge
